src/SocketManager.cpp

```diff
@@ -1,4 +1,5 @@
 #include "SocketManager.hpp"
+#include "signal.h"
```

Comment:
Can you move this into the `.hpp` file? As thats where all of our other library includes are